src/IB/IBFESurfaceMethod.cpp

```diff
@@ -303,60 +303,18 @@ IBFESurfaceMethod::preprocessIntegrateData(double current_time, double new_time,
 
     // Extract the FE data.
     d_X_vecs->copy("solution", { "current", "new", "half" });
-
-    d_U_systems.resize(d_num_parts);
-    d_U_current_vecs.resize(d_num_parts);
-    d_U_new_vecs.resize(d_num_parts);
-    d_U_half_vecs.resize(d_num_parts);
-
-    d_U_n_systems.resize(d_num_parts);
-    d_U_n_current_vecs.resize(d_num_parts);
-    d_U_n_new_vecs.resize(d_num_parts);
-    d_U_n_half_vecs.resize(d_num_parts);
-
-    d_U_t_systems.resize(d_num_parts);
-    d_U_t_current_vecs.resize(d_num_parts);
-    d_U_t_new_vecs.resize(d_num_parts);
-    d_U_t_half_vecs.resize(d_num_parts);
-
-    d_F_systems.resize(d_num_parts);
-    d_F_half_vecs.resize(d_num_parts);
-    d_F_IB_ghost_vecs.resize(d_num_parts);
+    d_U_vecs->copy("solution", { "current", "new", "half" });
```

Comment:
This pattern is getting pretty common --- I wonder if it is worth batching it up, e.g.,
```
batch_vec_copy( {d_X_vecs, d_U_vecs, d_U_n_vecs, d_U_t_vecs}, "solution", {"current", "new", "half"});
```

Reply:
Maybe - the IIM project adds a lot more systems, though, so I'd rather get more of that merged first before we add more utility functions.